src/Illuminate/Foundation/Console/KeyGenerateCommand.php

```diff
@@ -53,6 +59,25 @@ public function handle()
         $this->components->info('Application key set successfully.');
     }
 
+    /**
+     * Ask the user if they want to save the existing key.
+     */
+    protected function shouldSaveExistingKey(): bool
+    {
+        $currentKey = $this->laravel['config']['app.key'];
+
+        if (! empty($currentKey) && $this->confirm(
+            'There is already an app key. Do you want to store the old key before generating a new one?',
+            true
+        )) {
+            $this->call('key:rotate');
```

Comment:
I'd prefer not to have the exact action of the rotate inside method which is named "should*" and returns boolean.

Reply:
Tell me how you would like te naming please.

Reply:
Just move the call out of this method, it will become cleaner and will do only one job, determine Boolean flag.